pkg/app/server.go

```diff
@@ -145,6 +149,24 @@ func RunKubeStateMetrics(ctx context.Context, opts *options.Options) error {
 		}
 	}
 
+	kubeConfig, err := clientcmd.BuildConfigFromFlags(opts.Apiserver, opts.Kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to build config from flags: %v", err)
+	}
+
+	// Ensure cache prerequisites are set up only once.
+	if gvkMaps.Map == nil {
```

Comment:
we could use a global sync.Once for that purpose: https://pkg.go.dev/sync#Once

this would avoid having to pass down a nil gvkMaps from the wrapper